### ISSUE TITLE
Dao Considerations

### ISSUE THREAD

**Issue by nfurfaro** [unlock-protocol/locked.fyi#45]
A few details about the dao configuration and Aragon in general.

1.) The DAO will need to be both the lock beneficiary and a Lock Manager.
2.) We need to decide on the voting parameter configuration
3.) I'm not sure yet how to direct the dao to mint shares for authors. I'm sure this is fairly straightforward. Ideally, only the hook has this permission.
4.) The necessary Aragon components I've identified are:
  - Agent - replaces vault, handles token storage and calling external contracts
  - Tokens 
  - Voting
  - Redemptions
The [dandelion](https://1hive.org/dandelion) template has all of these out of the box, but also includes functionality we maybe don't need:
![image](https://user-images.githubusercontent.com/17053076/85080395-49327900-b186-11ea-96b4-df3a7e85953e.png)

Another option is to use a more basic template, and use the aragon-cli to add additional functionality as needed. For example, the [membership](https://rinkeby.aragon.org/#/create) template has everything I listed as needed except for redemptions. Not sure yet how difficult it is to take this approach yet.

Aragon has an app called "Aragon Agent" https://hack.aragon.org/docs/guides-use-agent
TLDR; This will essentially allow the Dao to call "withdraw", or otherwise configure the lock and/or hook.
>The Agent app (or Aragon Agent) is an Aragon app that can be installed in any Aragon DAO. It's main feature is its ability to perform arbitrary calls to contracts. This means it can be thought of as the external interface of a DAO.
Aragon Agent is a fully-fledged Ethereum account owned by an Aragon organization. It's like a multi-signature account on steroids that enables organizations to interact with any Ethereum contract or protocol.

(see #34 for context)

**Comment by nfurfaro** [comment]
Actually, re: 3.) above
looks like setting up permissions is really straightforward.
![image](https://user-images.githubusercontent.com/17053076/85080809-6287f500-b187-11ea-857c-ce0d3e44e828.png)
